[GR-1230] Send request to MISO

avarsava opened the pull request:
This is a draft because I want feedback before adding this feature to all the other reports. This is currently only implemented on Single-Lane TS. Looking for assistance in particular on the way the CSS causes the buttons to be slightly misaligned on the right side of the sidebar, and whether anything needs more documentation.

avarsava committed:
We need the requests library

avarsava committed:
Add MISO url to env

avarsava committed:
basically nothing

avarsava committed:
Draft of request dict

avarsava committed:
draft button

avarsava committed:
post the request, but this is probably wrong

avarsava committed:
Very shoddy demo, but it works

avarsava committed:
Dict format as requested by Dillan

avarsava committed:
Use ID rather than alias. needs to be parsed out of sample provenance id

avarsava committed:
It worksgit add -u

avarsava committed:
Your services won't be needed after all, requests

avarsava committed:
remove debug log

avarsava committed:
cleanup

avarsava committed:
Minimize # of new IDs

avarsava committed:
Move Form to util f'n

avarsava committed:
Utility f'n for update

avarsava committed:
Document the particular metric format

avarsava committed:
the rituals are complete. the css works

avarsava committed:
Don't show the QC in MISO button when there's no runs selected

avarsava committed:
good ids

avarsava committed:
follow your own rules varsava

avarsava commented on application/dash_application/assets/styles.css:
```diff
@@ -377,6 +373,13 @@ input[type="button"] {
+    float: left;
+    width: 30%;
+    margin-left: 2.5%;
+    margin-right: 2.5%;
```
I know 3(30) + 2(2.5) != 100, but when I set these to 5, they'd overflow, and i couldn't identify the margin or padding that was ruining it

apmasell replied:
CSS is supposed to make sense? If it bothers you, maybe `calc`. I'm good with it.

avarsava committed:
Remove imports from previous version

slazicoicr commented on application/dash_application/utility/df_manipulation.py:
```diff
@@ -475,3 +475,37 @@ def remove_suffixed_columns(df: DataFrame, suffix: str) -> DataFrame:
+    }]
+    Returns [json string dump of http body, button style]
+    """
+    if (len(df.index) > 0):
```
This is equivalent to `len(df) > 0`

avarsava replied:
it is, but len(df.index) is slightly more performant: https://stackoverflow.com/a/18317067

slazicoicr commented on application/dash_application/utility/df_manipulation.py:
```diff
@@ -475,3 +475,37 @@ def remove_suffixed_columns(df: DataFrame, suffix: str) -> DataFrame:
+            miso_request['library_aliquots'].append({
+                'name': split_provenance_id[2],
+                'metrics': metrics_this_row,
+                'run': split_provenance_id[0]
```
Would this require `lane` as well?

avarsava replied:
Shouldn't, this is the format @djcooke specified for receiving into MISO.
Dillan, should I be sending MISO the lane as well?

djcooke replied:
Depends how specific you want it to be:

* If you provide only the library aliquot, MISO will include all lanes that it has been run in
* If you provide run + library aliquot, MISO will filter to include only lanes from that run that include the library aliquot
* If you provide run + lane + library aliquot, MISO could filter to include only that lane

If the metrics are specific to a lane, then you probably want to provide the lane.

avarsava replied:
@slazicoicr Metrics are specific to a Lane because of how IUSes work, right?

avarsava replied:
Updated

slazicoicr commented on application/dash_application/views/single_lane_ts.py:
```diff
@@ -524,6 +527,15 @@ def update_pressed(click,
 
+        (miso_request, miso_button_style) = util.build_miso_info(df, title, 
+            [{
+                'title': 'Passed Filter Reads (*10^6)',
```
Should this be `Total Clusters`?

avarsava replied:
Looks like part of SLTS got overlooked when making this switch. I'll fix it in another PR then rebase this and fix

avarsava committed:
Fix Reads -> Clusters

avarsava committed:
Real endpoint name

avarsava committed:
Add MISO_URL to README

avarsava committed:
Add Lane Number

avarsava committed:
lane -> partition

avarsava committed:
Fix URL

avarsava committed:
run -> run_id

avarsava commented:
Demonstrated to work with the MISO half! time to expand to the other reports and open for review

avarsava committed:
All the Single-Lane reports

It seems like the cfMeDIP threshold works properly in MISO and NOT in Dashi. A problem for another ticket though

avarsava committed:
new tab'

avarsava committed:
changelog
